packages/eslint-plugin/docs/rules/consistent-return.md

```diff
@@ -0,0 +1,33 @@
+---
+description: 'Require `return` statements to either always or never specify values.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/consistent-return** for documentation.
+
+This rule extends the base [`eslint/consistent-return`](https://eslint.org/docs/rules/consistent-return) rule.
+This version adds support for functions that return void type.
+
+<!--tabs-->
+
+#### ❌ Incorrect
+
+```ts
+function foo(): undefined {}
+function bar(flag: boolean): undefined {
+  if (flag) return foo();
+  return;
+}
+```
+
+### ✅ Correct
+
+```ts
+/* eslint @typescript-eslint/consistent-return: "error" */
```

Comment:
[nit + asking] `❌ Incorrect` section doesn't have similar eslint-enable directive. Perhaps we should keep them in sync?

Only two rules contain eslint-enable directives in their docs: [`no-implied-eval`](https://github.com/typescript-eslint/typescript-eslint/blob/8622286207b3e04ddd260eaa8ac71927d47edd7d/packages/eslint-plugin/docs/rules/no-implied-eval.md), [`default-param-last`](https://github.com/typescript-eslint/typescript-eslint/blob/8622286207b3e04ddd260eaa8ac71927d47edd7d/packages/eslint-plugin/docs/rules/default-param-last.md). So it is not widely used in our docs

From my point of view, the rule documentation already implies that in all examples the rule is enabled

What do you think?

Reply:
Yeah if most docs don't do it then this one shouldn't. I'd consider it a docs bug that those two docs pages do have the comment.

Discussing adding them in the way the core ESLint docs do would be a separate issue from this.